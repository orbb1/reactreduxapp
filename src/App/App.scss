.App-container {
    text-align: center;
    width: 320px;
    margin: auto;
    border-bottom: 5px solid grey;
}