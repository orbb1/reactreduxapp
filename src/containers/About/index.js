import React, {Component} from 'react';

import './About.scss';

class About extends Component {
    render() {
        return (
            <div className="About-Container container">
                <h4>This is my simple todo app</h4>
                <p>I made this page for practising React and Redux</p>
                <hr/>
                <p>Link to <a href="https://github.com/orbb1/reactreduxapp">GitHub repo</a></p>
            </div>
        )
    }
}

export default About;