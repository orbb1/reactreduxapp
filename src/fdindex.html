<html>
    <head>
        <title>New page 1</title>
        <style>
            
        </style>
    </head>
    <body>
        <h1>Hello World</h1>
        <input type="text"/>
        <p>Paragraph</p>
        <p>Paragraph 2</p>
        <span>Span</span>
        <span>Span 2</span>
    </body>
</html>