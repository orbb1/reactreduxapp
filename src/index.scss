@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500');

body {
  margin: 0;
  padding: 0;
  font-family: 'Roboto', sans-serif;
  margin-left: calc(100vw - 100%);
  .container {
    text-align: center;
    width: 400px;
    margin: auto;
  }
}



@media (min-width: 320px) and (max-width: 480px) {
  .container {
    width: 300px;
  }
}