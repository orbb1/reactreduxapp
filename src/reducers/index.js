import {combineReducers} from 'redux';

import tasks from './tasks';
import taskFilter from './taskFilter';
import asyncActions from './asyncActions';

export default combineReducers({
    tasks,
    taskFilter,
    asyncActions
})