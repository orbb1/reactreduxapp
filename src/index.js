import ReactDOM from 'react-dom';
import React from 'react';
import {App} from './App';

const root = document.getElementById('root');

ReactDOM.render(<App/>, root);