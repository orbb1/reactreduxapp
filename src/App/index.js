import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducer from '../reducers';
import thunk from 'redux-thunk';
import { Route, BrowserRouter } from 'react-router-dom';
import logger from 'redux-logger';
import "./App.scss";

// TapEventPlugin for Material Ui
import injectTapEventPlugin from 'react-tap-event-plugin'
injectTapEventPlugin();

//Material Ui
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import TaskList from '../containers/TaskList/index';
import About from '../containers/About/index';
import { Nav } from '../components/Nav';

let middleware = [logger, thunk]
const store = createStore(reducer, composeWithDevTools(applyMiddleware(...middleware)));

export const App = () => (
  <Provider store={store}>
    <BrowserRouter>
      <MuiThemeProvider>
        <div className="App-container">
          <Nav/>
          <Route exact path="/" component={TaskList}/>
          <Route path="/about" component={About}/>
        </div>
      </MuiThemeProvider>
    </BrowserRouter>
  </Provider>);