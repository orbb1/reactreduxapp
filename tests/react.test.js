import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import About from '../src/containers/About';
import {App} from '../src/App';

it('reanders App', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
});

it('snapshot App', () => {
    const app = renderer.create(<App/>);
    app.toJSON();
    expect(app).toMatchSnapshot();
})

it('snapshot About container', () => {
    const aboutPage = renderer.create(<About></About>);
    aboutPage.toJSON();
    expect(aboutPage).toMatchSnapshot();
});