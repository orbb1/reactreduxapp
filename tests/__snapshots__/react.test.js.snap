exports[`test snapshot About container 1`] = `
<div
  className={undefined}
  style={undefined}>
  <div
    style={
      Object {
        "transform": "translateX(100%)",
      }
    }>
    <div
      className="About-Container container">
      <h4>
        This is my simple todo app
      </h4>
      <p>
        I made this page for practising React and Redux
      </p>
      <hr />
      <p>
        Link to 
        <a
          href="https://github.com/orbb1/reactreduxapp">
          GitHub repo
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`test snapshot App 1`] = `
<div>
  <div
    className="Nav-container container">
    <div
      className="Nav-item">
      <a
        href="/"
        onClick={[Function]}>
        <strong>
          Tasklist
        </strong>
      </a>
    </div>
    <div
      className="Nav-item">
      <a
        href="/about"
        onClick={[Function]}>
        About
      </a>
    </div>
  </div>
  <div
    className={undefined}
    style={undefined}>
    <div
      style={
        Object {
          "transform": "translateX(-100%)",
        }
      }>
      <div
        className="container">
        <div
          className="TaskList-form">
          <form
            onSubmit={[Function]}>
            <div
              className="el-inline"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "cursor": "auto",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "height": 48,
                  "lineHeight": "24px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": 256,
                }
              }>
              <div
                style={
                  Object {
                    "bottom": 12,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "muiPrepared": true,
                    "opacity": 1,
                    "position": "absolute",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }>
                Min 3 characters
              </div>
              <input
                disabled={false}
                id="undefined-Min3characters-undefined-640"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "WebkitTextFillColor": "rgba(0, 0, 0, 0.87)",
                    "appearance": "textfield",
                    "backgroundColor": "rgba(0,0,0,0)",
                    "border": "none",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "inherit",
                    "font": "inherit",
                    "height": "100%",
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
                }
                type="text" />
              <div>
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottom": "solid 1px",
                      "borderColor": "#e0e0e0",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "width": "100%",
                    }
                  } />
                <hr
                  aria-hidden="true"
                  style={
                    Object {
                      "borderBottom": "solid 2px",
                      "borderColor": "#00bcd4",
                      "borderLeft": "none",
                      "borderRight": "none",
                      "borderTop": "none",
                      "bottom": 8,
                      "boxSizing": "content-box",
                      "margin": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "transform": "scaleX(0)",
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  } />
              </div>
            </div>
            <div
              className={undefined}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 2,
                  "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                         0 1px 4px rgba(0, 0, 0, 0.12)",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "minWidth": 88,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }>
              <button
                disabled={false}
                href={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "#ffffff",
                    "border": 10,
                    "borderRadius": 2,
                    "boxSizing": "border-box",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "height": 36,
                    "lineHeight": "36px",
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "textAlign": "center",
                    "textDecoration": "none",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "verticalAlign": null,
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
                tabIndex={0}
                type="submit">
                <div
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}>
                  <div
                    style={
                      Object {
                        "backgroundColor": false,
                        "borderRadius": 2,
                        "height": 36,
                        "muiPrepared": true,
                        "top": 0,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }>
                    <span
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontSize": 14,
                          "fontWeight": 500,
                          "letterSpacing": 0,
                          "margin": 0,
                          "muiPrepared": true,
                          "opacity": 1,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "position": "relative",
                          "textTransform": "uppercase",
                          "userSelect": "none",
                        }
                      }>
                      Add task
                    </span>
                  </div>
                </div>
              </button>
            </div>
          </form>
        </div>
        <div
          className="TaskList-form">
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 48,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }>
            <div
              style={
                Object {
                  "bottom": 12,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "muiPrepared": true,
                  "opacity": 1,
                  "position": "absolute",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }>
              Filter tasks
            </div>
            <input
              disabled={false}
              id="undefined-Filtertasks-undefined-62271"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "WebkitTextFillColor": "rgba(0, 0, 0, 0.87)",
                  "appearance": "textfield",
                  "backgroundColor": "rgba(0,0,0,0)",
                  "border": "none",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "100%",
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "width": "100%",
                }
              }
              type="text" />
            <div>
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottom": "solid 1px",
                    "borderColor": "#e0e0e0",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "width": "100%",
                  }
                } />
              <hr
                aria-hidden="true"
                style={
                  Object {
                    "borderBottom": "solid 2px",
                    "borderColor": "#00bcd4",
                    "borderLeft": "none",
                    "borderRight": "none",
                    "borderTop": "none",
                    "bottom": 8,
                    "boxSizing": "content-box",
                    "margin": 0,
                    "muiPrepared": true,
                    "position": "absolute",
                    "transform": "scaleX(0)",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "width": "100%",
                  }
                } />
            </div>
          </div>
        </div>
        <div>
          <div
            className={undefined}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#ffffff",
                "borderRadius": 2,
                "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                       0 1px 4px rgba(0, 0, 0, 0.12)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "minWidth": 88,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }>
            <button
              disabled={false}
              href={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "#ffffff",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
              tabIndex={0}
              type="button">
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}>
                <div
                  style={
                    Object {
                      "backgroundColor": false,
                      "borderRadius": 2,
                      "height": 36,
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }>
                  <span
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontSize": 14,
                        "fontWeight": 500,
                        "letterSpacing": 0,
                        "margin": 0,
                        "muiPrepared": true,
                        "opacity": 1,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "position": "relative",
                        "textTransform": "uppercase",
                        "userSelect": "none",
                      }
                    }>
                    Get tasks
                  </span>
                </div>
              </div>
            </button>
          </div>
        </div>
        <ul
          className="TaskList-tasks" />
      </div>
    </div>
  </div>
</div>
`;
